src/main/java/mate/academy/Main.java

```diff
@@ -1,7 +1,34 @@
 package mate.academy;
 
+import java.math.BigDecimal;
+import mate.academy.dao.BookDao;
+import mate.academy.lib.Injector;
+import mate.academy.model.Book;
+import mate.academy.service.BookService;
+import mate.academy.service.impl.BookServiceImpl;
+
 public class Main {
+    private static final Injector injector = Injector.getInstance("mate.academy");
+
     public static void main(String[] args) {
+        BookDao bookDao = (BookDao) injector.getInstance(BookDao.class);
+        BookService bookService = new BookServiceImpl(bookDao);
```

Comment:
Instantiating `BookServiceImpl` with `new` keyword here is not consistent with the dependency injection principle. You should retrieve the instance from the `Injector`.